### ISSUE TITLE
False positive for the `a11y-no-noninteractive-element-to-interactive-role` a11y audit

### ISSUE THREAD

**Issue by HiDeoo** [withastro/astro#10346]
### Astro Info

```block
Astro                    v4.4.12
Node                     v20.11.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             @astrojs/starlight
```

### Describe the Bug

The `a11y-no-noninteractive-element-to-interactive-role` Dev Toolbar a11y audit seems to report false positives when using an interactive `role` attribute on elements that should be considered interactive.

For example, the following simplified code from Starlight [`<Tabs>` component](https://github.com/withastro/starlight/blob/86044429a699d8eb6fc5c92871f2e7d5026ac05e/packages/starlight/user-components/Tabs.astro) will trigger the warning:

```astro
<a role="tab" href={'#' + panelId}>{label}</a>
```

The rule selector is [`[role]:not(${interactiveElements.join(',')})`](https://github.com/withastro/astro/blob/a2e9b2b936666b2a4779feb00dcb8ff0ab82c2ec/packages/astro/src/runtime/client/dev-toolbar/apps/audit/a11y.ts#L449) with `interactiveElements` being [`['button', 'details', 'embed', 'iframe', 'label', 'select', 'textarea']`](https://github.com/withastro/astro/blob/a2e9b2b936666b2a4779feb00dcb8ff0ab82c2ec/packages/astro/src/runtime/client/dev-toolbar/apps/audit/a11y.ts#L45C29-L45C100).

Altho, according to this [MDN reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#interactive_content) (note that the anchor does seems to always work as expected), the "Interactive content" section lists the elements from the `interactiveElements` array as well as some elements that belong to this category only under specific conditions:

> `<a>`, if the `href` attribute is present
> `<audio>`, if the `controls` attribute is present
> `<img>`, if the `usemap` attribute is present
> `<input>`, if the `type` attribute is not in the hidden state
> `<object>`, if the `usemap` attribute is present
> `<video>`, if the `controls` attribute is present

This second list of elements does not seem to be considered by the rule.

### What's the expected result?

I expected the warning to not be triggered when using the `tab` interactive role on an anchor element with an `href` attribute which makes it interactive.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-xcr9d3?file=src%2Fcontent%2Fdocs%2Findex.mdx

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
Closed by https://github.com/withastro/astro/pull/10343
